pkg/controller/gitopssyncresc/gitopssyncresc_controller.go

```diff
@@ -0,0 +1,518 @@
+		// Add app name to map
+		clusterAppsetNameMap := managedClusterAppNameMap[reportKey]
+		if len(clusterAppsetNameMap) == 0 {
+			clusterAppsetNameMap = make(map[string]string)
+			managedClusterAppNameMap[reportKey] = clusterAppsetNameMap
+		}
+
+		clusterAppsetNameMap[managedClusterName] = appNs + "_" + appName
+
+		if !eCond.MatchString(k) && !wCond.MatchString(k) {
+			continue
+		}
+
+		repCond := &appsetreport.Condition{
+			Type:    k[10:],
+			Message: v.(string),
+		}
+
+		if repCond.Type == "OperationError" {
+			operationStateError = repCond
+		} else {
+			reportConditions = append(reportConditions, *repCond)
+		}
+	}
+
+	// Check for duplicate operation state in condition before adding it
+	if operationStateError != nil {
+		foundCondition := false
+
+		for _, con := range reportConditions {
+			opStateReason := ""
+
+			index := strings.Index(con.Message, "reason:")
+			if index != -1 {
+				opStateReason = con.Message[index : len(con.Message)-3]
+			}
+
+			if opStateReason != "" && strings.Contains(operationStateError.Message, opStateReason) {
+				foundCondition = true
+
+				break
+			}
+		}
+
+		if !foundCondition {
+			reportConditions = append(reportConditions, *operationStateError)
+		}
+	}
+
+	clusterConditions := report.Statuses.ClusterConditions
+	if len(clusterConditions) == 0 {
+		clusterConditions = make([]appsetreport.ClusterCondition, 0)
+	}
+
+	clusterCondition := appsetreport.ClusterCondition{
+		Cluster:    managedClusterName,
+		Conditions: reportConditions,
+	}
+	clusterConditions = append(clusterConditions, clusterCondition)
+	report.Statuses.ClusterConditions = clusterConditions
+
+	return nil
+}
+
+func (r *GitOpsSyncResource) writeAppSetResourceFile(report *appsetreport.MulticlusterApplicationSetReport) error {
+	reportJSON, err := yaml.Marshal(report)
+	if err != nil {
+		return err
+	}
+
+	reportName := filepath.Join(r.ResourceDir, report.Name+".yaml")
+	klog.Info(fmt.Sprintf("writing appset report: %v", reportName))
+
+	if err := ioutil.WriteFile(reportName, reportJSON, 0600); err != nil {
+		klog.Error(err, fmt.Sprintf("failed to write appset report yaml file: %v", reportName))
+		return err
+	}
+
+	return nil
+}
+
+func getResourceMapList(related []interface{}, cluster string) []appsetreport.ResourceRef {
+	resourceList := make([]appsetreport.ResourceRef, 0)
+
+	for _, rel := range related {
+		if relatedmap, ok := rel.(map[string]interface{}); ok {
+			klog.V(4).Info(fmt.Sprintf("related: %v", relatedmap))
+
+			if slices.Contains(ExcludeResourceList, relatedmap["kind"].(string)) {
+				klog.V(4).Infof("skip resource kind: %v", relatedmap["kind"].(string))
+				continue
+			}
+
+			if relatedItems, ok := relatedmap["items"].([]interface{}); ok {
+				for _, r := range relatedItems {
+					relatedItem := r.(map[string]interface{})
+					if itemCluster, ok := relatedItem["cluster"]; !ok || itemCluster != cluster {
+						klog.V(4).Infof("skip resource from cluster:%v", itemCluster)
+						continue
```

Comment:
@philipwu08 @lennysgarage  It is a better practice to print out the message without verbose mode before `return` or `continue`. Then we can get these important messages from the original pod log during troubleshooting. Or we will have to get the info by reseting the verbose mode and restarting the controller

```
if itemCluster, ok := relatedItem["cluster"]; !ok || itemCluster != cluster {
    klog.V(4).Infof("skip resource from cluster:%v", itemCluster)
    continue
```
====> 
```
if itemCluster, ok := relatedItem["cluster"]; !ok || itemCluster != cluster {
    klog.Infof("skip resource from cluster:%v", itemCluster)
    continue
```

Also we just use debug level 1 klog.V(1) to print out the additional debug info. 
```
klog.V(4).Info(fmt.Sprintf("related: %v", relatedmap))
====>
klog.V(1).Info(fmt.Sprintf("related: %v", relatedmap))

```
Could you go through your controllers and adjust the klog verbose mode levels accordingly?

Reply:
Updated sync resc controller.

Reply:
Updated the aggregation controller.